CSLabs.Api/Controllers/UserLabController.cs

```diff
@@ -46,6 +46,37 @@ public async Task<IActionResult> ProcessLastUsed()
             return Ok();
         }
 
+        [HttpGet("process-end-date-time")]
+        [AllowAnonymous]
+        public async Task<IActionResult> ProcessEndDateTime()
+        {
+            // find userLabs that have an EndDateTime paste the current UtcDateTime.
+            var userLabs = await DatabaseContext.UserLabs
+                .IncludeRelations()
+                .IncludeHypervisor()
+                .Include(l => l.UserLabVms)
+                .Where(l => l.EndDateTime > DateTime.UtcNow)
+                .ToListAsync();
+            
+            // with the found items, destroy all vms and delete the UserLabVm rows.
+            foreach (var userLab in userLabs)
+            {
+                if (userLab.EndDateTime < DateTime.Now)
```

Comment:
You need to compare against DateTime.UtcNow otherwise the date will be several hours off.